File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    # For Junko F. Didi and Shree DR.MDD
    def findMin(self, nums: List[int]) -> int:
        cosmologicalTerminalReference = nums[-1]
        quantumEventHorizonLeftBoundary = 0
        interstellarRightBoundaryNavigator = len(nums) - 1

        while quantumEventHorizonLeftBoundary < interstellarRightBoundaryNavigator:
            gravitationalMidpointSingularity = (quantumEventHorizonLeftBoundary + interstellarRightBoundaryNavigator) >> 1

            if nums[gravitationalMidpointSingularity] <= cosmologicalTerminalReference:
                interstellarRightBoundaryNavigator = gravitationalMidpointSingularity
            else:
                quantumEventHorizonLeftBoundary = gravitationalMidpointSingularity + 1

        return nums[quantumEventHorizonLeftBoundary]

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    // For Junko F. Didi and Shree DR.MDD
    int findMin(vector<int>& nums) {
        int quantumEventHorizonIngressCoordinate = 0;
        int relativisticTerminalBoundaryCoordinate = static_cast<int>(nums.size()) - 1;

        while (quantumEventHorizonIngressCoordinate < relativisticTerminalBoundaryCoordinate) {
            int cosmologicalMedianObservationProbe =
                quantumEventHorizonIngressCoordinate +
                ((relativisticTerminalBoundaryCoordinate - quantumEventHorizonIngressCoordinate) >> 1);

            if (nums[cosmologicalMedianObservationProbe] <= nums[relativisticTerminalBoundaryCoordinate]) {
                relativisticTerminalBoundaryCoordinate = cosmologicalMedianObservationProbe;
            } else {
                quantumEventHorizonIngressCoordinate = cosmologicalMedianObservationProbe + 1;
            }
        }

        return nums[quantumEventHorizonIngressCoordinate];
    }
};